### PR TITLE
Support for scanning of an OCI image given a tag.

### PR DESCRIPTION
**Pull request by shimish2**
Hi @knqyf263 
This PR will support the feature for scanning an OCI image given a tag, currently it only support first tag.

Condition:- Specify an image name and tag ":" separated.
If correct image name and tag is specified ":" separated, image with given tag will be return otherwise first one will be return.

I have not added test cases for this change, please review this changes and I will add a test code for the same.

### CODE REVIEW

**Comment by shimish2** [comment]
#120

File: image/image.go [hunk 1]
```diff
@@ -186,7 +188,21 @@ func fileOpener(fileName string) func() (io.ReadCloser, error) {
 }
 
 func tryOCI(fileName string) (v1.Image, error) {
-	lp, err := layout.FromPath(fileName)
+	var inputTag string
+
+	var inputFileName string
```
Comment (knqyf263) [inline image/image.go:193]: nit
```suggestion
	var inputTag, inputFileName string
```

File: image/image.go [hunk 2]
```diff
@@ -205,11 +221,37 @@ func tryOCI(fileName string) (v1.Image, error) {
...
+					return nil, xerrors.New("invalid OCI image")
+				}
+
+				break
+			}
+		}
+
+		if img == nil {
+			return nil, xerrors.New("invalid OCI image tag")
+		}
```
Comment (knqyf263) [inline image/image.go:247]: Could you define a new function for these processes?

Comment (shimish2) [reply image/image.go:247]: Sure, I will define one

**Comment by shimish2** [comment]
@knqyf263 could you please let me know why benchmark is failing and what needs to be done from my side to pass benchmark test, I am not able to run locally due to some tcp timeout issue.

**Comment by knqyf263** [comment]
Hmm. It looks like this change doesn't affect performance. Let me re-run the job.

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@shimish2 Thanks!

**Pull request merged**